baton-maven-plugin/src/main/java/org/technologybrewery/baton/BatonMojo.java

```diff
@@ -0,0 +1,236 @@
+
+    private Map<String, MigrationTarget> migrations = new HashMap<>();
+
+    protected ObjectMapper initializeObjectMapper() {
+        ObjectMapper localObjectMapper = new ObjectMapper();
+
+        SimpleModule module = new SimpleModule();
+        module.addAbstractTypeMapping(ValidatedElement.class, MigrationTarget.class);
+
+        localObjectMapper.registerModule(module);
+
+        return localObjectMapper;
+    }
+
+    protected void defaultFileSets() {
+        if (fileSets == null) {
+            getLog().debug("Defining default file set...");
+
+            List<FileSet> defaultFileSets = new ArrayList<>();
+            // add source:
+            FileSet sourceFileSet = new FileSet();
+            sourceFileSet.setDirectory(sourceDirectory.getAbsolutePath());
+            defaultFileSets.add(sourceFileSet);
+            getLog().debug(String.format("\tAdded: %s", sourceFileSet.getDirectory()));
+
+            // add test:
+            if (testDirectory != null) {
+                FileSet testFileSet = new FileSet();
+                testFileSet.setDirectory(testDirectory.getAbsolutePath());
+                defaultFileSets.add(testFileSet);
+                getLog().debug(String.format("\tAdded: %s", testFileSet.getDirectory()));
+            }
+
+            // add common root files:
+            FileSet rootFileSets = new FileSet();
+            rootFileSets.setDirectory(baseDirectory.getAbsolutePath());
+            rootFileSets.addInclude("pom.xml");
+            rootFileSets.addInclude("*.toml");
+            defaultFileSets.add(rootFileSets);
+            getLog().debug(String.format("\tAdded: %s with includes: %s", rootFileSets.getDirectory(), rootFileSets.getIncludes()));
+
+            fileSets = defaultFileSets.toArray(new FileSet[0]);
+
+            getLog().debug("Default file set definition completed");
+        }
+    }
+
+    @Override
+    public void execute() throws MojoExecutionException, MojoFailureException {
+        loadMigrations();
+
+        defaultFileSets();
+
+        BatonExecutionSummary summary = performMigration(migrations.values());
+        getLog().info(summary.getSummary());
+
+    }
+
+    BatonExecutionSummary performMigration(Collection<MigrationTarget> targets) {
+        BatonExecutionSummary executionSummary = new BatonExecutionSummary();
+        for (MigrationTarget target : targets) {
+            try {
+                getLog().debug(String.format("Executing Migration: %s (%s)", target.getName(), target.getImplementation()));
+                Class<Migration> implementationClass = (Class<Migration>) Class.forName(target.getImplementation());
+                Constructor<Migration> constructor = implementationClass.getConstructor();
+                Migration migration = constructor.newInstance();
+                FileSet[] migrationSpecificFileSets = (CollectionUtils.isNotEmpty(target.getFileSets())) ? getFileSetsForTarget(target) : fileSets;
```

Comment:
I feel like it's worth calling out in the README that migration-specific filesets directly replace/override the plugin configuration filesets.

Reply:
Added the following as well as a section in the README specifically about the `migrations.json` configuration options.

> **NOTE:** File set configurations specified in your `migrations.json` will override those specified to your plugin at
large.